src/cli/index.js

```diff
@@ -537,6 +537,7 @@ export function main({
 }
 
 async function start(): Promise<void> {
+  prepareEnv();
```

Comment:
I think this should be done in `getRcConfigForCwd` instead of changing the whole `process.env`. I don't think it is a good idea to patch/mutate `process.env`

Reply:
Normally, I'd agree. In this case, though, I figured that setting HOME only if it's not already set should be safe (the fact that NPM does it could be evidence for its safety?), and, since Yarn is promoted as a drop-in replacement for NPM, I figured there's benefit to doing what NPM does. (My thinking here is probably colored by my running into #1538.)

I'm happy to go either way, though, so I can change the PR this evening to use `getRcConfigForCwd`.

Thanks for the feedback!

Reply:
Upon further investigation, I don't think that `getRcConfigForCwd` is correct. That finds config files but doesn't process them. The environment variable changes instead need to be made within `NpmRegistry`. I just pushed an updated PR with these changes.